### PR TITLE
Feature/general oidc

### PR DESCRIPTION
**Pull request by jasperweyne**

### CODE REVIEW

**Comment by A-Daneel** [comment]
tests are now passing, lets do some user testing just in case and merge this

File: tests/Functional/Controller/Admin/SecurityControllerTest.php [hunk 1]
```diff
@@ -91,7 +102,16 @@ public function testDeleteAction(): void
...
+        $localUser = $this->em->getRepository(LocalAccount::class)->findAll()[0];
+        $id = $localUser->getId();
+
+        // Act
+        $crawler = $this->client->request('GET', "/admin/security/{$id}/roles");
+        $this->assertEquals(200, $this->client->getResponse()->getStatusCode());
+        $form = $crawler->selectButton('Opslaan')->form();
+        $form['form[admin]']->setValue(false);
+        $this->client->submit($form);
+        $this->assertSelectorTextContains('.container', 'Rollen bewerkt');
```
Comment (jasperweyne) [inline tests/Functional/Controller/Admin/SecurityControllerTest.php:115]: This should also check whether the roles actually changed in database

File: src/Entity/Security/LocalAccount.php [hunk 2]
```diff
@@ -91,9 +100,9 @@ public function setId(string $id): self
      * @see
      * @since
      */
-    public function getEmail(): string
+    public function getEmail(): ?string
     {
-        return (string) $this->email;
+        return '' != $this->email ? $this->email : null;
```
Comment (jasperweyne) [inline src/Entity/Security/LocalAccount.php:105]: Wot?

File: tests/Functional/Controller/Admin/SecurityControllerTest.php (hunk 1 above)
Comment (A-Daneel) [reply tests/Functional/Controller/Admin/SecurityControllerTest.php:115]: true, changed it to check for that as well

**Pull request merged**